src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/histogram.js

```diff
@@ -0,0 +1,711 @@
+                        )
+                    case 'wheel':
+                        return d3.event.button === scroll_wheel
+                    default:
+                        return false
+                }
+            })
+            .scaleExtent(extent)
+            .extent([
+                [0, 0],
+                [canvas.inner.width, canvas.inner.height],
+            ])
+            .on(zoomClass, zoomed)
+
+        const zoomElem = this.svg
+            .append('g')
+            .attr('class', zoomClass)
+            .attr('width', canvas.inner.width)
+            .attr('height', canvas.inner.height)
+            .style('fill', 'none')
+            .style('pointer-events', 'all')
+            .call(zoom)
+
+        const self = this
+        /**
+         * Helper function called on pan/scroll.
+         */
+        function zoomed() {
+            self.scale.zoom = d3.event.transform.k
+            let tmpScale = Object.assign({}, self.scale)
+            tmpScale.x = d3.event.transform.rescaleX(self.scale.x)
+            self.rescale(tmpScale, false)
+        }
+
+        return { zoomElem, zoom }
+    }
+
+    /**
+     * Initialise brushing functionality on the visualization.
+     *
+     * Brush is a tool which enables user to select points, and zoom into selection via
+     * keyboard shortcut or button event.
+     */
+    initBrushing(selectedZoomBtn, zoom) {
+        let extent
+        const brushClass = 'brush'
+
+        const brush = d3
+            .brushX()
+            .extent([
+                [0, 0],
+                [this.canvas.inner.width, this.canvas.inner.height],
+            ])
+            .on('start ' + brushClass, updateChart)
+
+        // The brush element must be child of zoom element - this is only way we found to have both
+        // zoom and brush events working at the same time. See https://stackoverflow.com/a/59757276 .
+        const brushElem = zoom.zoomElem
+            .append('g')
+            .attr('class', brushClass)
+            .call(brush)
+
+        const self = this
+
+        /**
+         * Zooms into selected fragment of plot.
+         *
+         * Based on https://www.d3-graph-gallery.com/graph/interactivity_brush.html
+         * Section "Brushing for zooming".
+         */
+        const zoomIn = () => {
+            const xMin = self.scale.x.invert(extent[0])
+            const xMax = self.scale.x.invert(extent[1])
+
+            self.scale.x.domain([xMin, xMax])
+            const dx = extent[1] - extent[0]
+            self.scale.zoom = self.scale.zoom * (self.canvas.inner.width / dx)
+
+            self.rescale(self.scale, true)
+        }
+
+        const zoomInKeyEvent = (event) => {
+            if (shortcuts.zoomIn(event)) {
+                zoomIn()
+                endBrushing()
+            }
+        }
+
+        /**
+         * Updates plot when brushing.
+         */
+        function updateChart() {
```

Comment:
I believe that function name should contain something about brushing if these docs are valid